ruby/ql/lib/codeql/ruby/frameworks/rack/internal/App.qll

```diff
@@ -86,4 +87,22 @@ module App {
     /** Gets a response returned from this request handler. */
     RP::PotentialResponseNode getAResponse() { result = resp }
   }
+
+  /** A read of the query string via `env['QUERY_STRING']`. */
+  private class EnvQueryStringRead extends Http::Server::RequestInputAccess::Range {
+    EnvQueryStringRead() {
+      exists(RequestHandler handler, DataFlow::ParameterNode env, ConstantValue key |
+        handler.getEnv() = env
+      |
+        this = env.getAnElementRead(key) and
+        key.isStringlikeValue("QUERY_STRING")
+      )
```

Comment:
```suggestion
      this =
        any(RequestHandler h)
            .getEnv()
            .getAnElementRead(ConstantValue::fromStringlikeValue("QUERY_STRING"))
```
Personally I find chaining code easier to read than having multiple local variables.

Reply:
Agreed, this is much clearer.